sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/DefaultAzureMonitorExporterOptions.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Monitor.OpenTelemetry.Exporter.Internals.Diagnostics;
+using Azure.Monitor.OpenTelemetry.Exporter.Internals.Platform;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter
+{
+    internal class DefaultAzureMonitorExporterOptions : IConfigureOptions<AzureMonitorExporterOptions>
+    {
+        private const string AzureMonitorExporterSectionFromConfig = "AzureMonitorExporter";
+        private readonly IConfiguration? _configuration;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultAzureMonitorExporterOptions"/> class.
+        /// </summary>
+        /// <param name="configuration"><see cref="IConfiguration"/> from which configuration for Azure Monitor can be retrieved.</param>
+        public DefaultAzureMonitorExporterOptions(IConfiguration? configuration = null)
+        {
+            _configuration = configuration;
+        }
+
+        public void Configure(AzureMonitorExporterOptions options)
```

Comment:
One issue with the design here. `IConfigureOptions` is going to execute in the order registered.

If user registers their own delegate before calling the extension...

```csharp
builder.Services.Configure<AzureMonitorExporterOptions>(o => o.ConnectionString = "mine");
builder.Services.AddOpenTelemetry(b => b.UseAzureMonitorExporter());
```

The user's changes will be potentially overridden by this class.

The only way I'm aware of to solve that is to register a custom `IOptionsFactory` for `AzureMonitorExporterOptions` which does the initialization (sets the defaults) prior to invoking any registered configurations.

Reply:
I will address it in a follow-up PR.